test/note_display.test.js

```diff
@@ -0,0 +1,530 @@
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+describe('note display criteria', function () {
+  before(async function () {
+    const crypto = new SNWebCrypto();
+    Uuid.SetGenerators(
+      crypto.generateUUIDSync,
+      crypto.generateUUID
+    );
+  });
+
+  beforeEach(function () {
+    this.modelManager = new PayloadManager();
+    this.itemManager = new ItemManager(this.modelManager);
+
+    this.createNote = async () => {
+      return this.itemManager.createItem(
+        ContentType.Note,
+        {
+          title: 'hello',
+          text: 'world'
+        }
+      );
+    };
+
+    this.createTag = async (notes = [], title = 'thoughts') => {
+      const references = notes.map((note) => {
+        return {
+          uuid: note.uuid,
+          content_type: note.content_type
+        };
+      });
+      return this.itemManager.createItem(
+        ContentType.Tag,
+        {
+          title: title,
+          references: references
+        }
+      );
+    };
+  });
+
+  const emptyCollection = function () {
+    const collection = new ItemCollection();
+    collection.setDisplayOptions(
+      ContentType.Note,
+      CollectionSort.CreatedAt,
+      'asc'
+    );
+    return collection;
+  }
+
+  const collectionWithNotes = function (titles = [], bodies = []) {
+    const collection = emptyCollection();
+    const notes = [];
+    titles.forEach((title, index) => {
+      notes.push(CreateItemFromPayload(Factory.createNotePayload(title, bodies[index])));
+    })
+    collection.set(notes);
+    return collection;
+  }
+
+  it('display criteria copying', async function () {
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeArchived = true;
+    });
+    const copy = NotesDisplayCriteria.Copy(criteria);
+    expect(copy.includeArchived).to.equal(true);
+  });
+
+  it('string query title', async function () {
+    const query = "foo";
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.searchQuery = { query: query };
+    });
+    const collection = collectionWithNotes(['hello', 'fobar', 'foobar', 'foo']);
+    expect(notesMatchingCriteria(criteria, collection).length).to.equal(2);
+  });
+
+  it('string query text', async function () {
+    const query = "foo";
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.searchQuery = { query: query };
+    });
+    const collection = collectionWithNotes(
+      [undefined, undefined, undefined, undefined],
+      ['hello', 'fobar', 'foobar', 'foo']
+    );
+    expect(notesMatchingCriteria(criteria, collection).length).to.equal(2);
+  });
+
+  it('string query title and text', async function () {
+    const query = "foo";
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.searchQuery = { query: query };
+    });
+    const collection = collectionWithNotes(['hello', 'foobar'], ['foo', 'fobar']);
```

Comment:
I think renaming `fobar` to something that explicitly does not look like `foo` would make this easier to read 😅

Reply:
I wanted to trick the search algorithm into trying to think fobar was == foobar 😆